scripts: Add updater script (implement #2)

Osyotr opened the pull request:
Нужно проверить работу и подумать над обновлением gt_knowledge_base. Обновлять master или ту ветку, на которой сейчас пользователь? А может обе? А может вообще не обновлять, в гайте написано, как самому обновлять.
Пока не ребазируйте, нужно с вопросами разобраться.

Osyotr committed:
scripts: Add updater script (implement #2)

blrB commented:
все интерфейсные проекты надо будет еще пересобрать grunt build

ну переключаться на мастер и обновлять его выглядит не логично, когда пользователь работает в своей ветке, обновить текущие ветки выглядит более разумно

KovalM commented:
Обе ветки будет логично, он останется работать в своей, но у него будет еще обнавленный мастер

Osyotr commented:
@KovalM хорошо, тогда еще git stash добавлю, на случай, если на отдельной ветке есть изменения

blrB commented:
@Osyotr 
sc-web
gt-ostis-drawings
set-ostis-drawings master
web-scn-editor

требуют grunt build

sc-machine требует make_all.sh

Osyotr commented:
Если есть кто-нибудь, кому нужно обновить базу, проверьте.

Теперь если пользователь не на ветке "master", то все его изменения на своей ветке временно сохраняются, идёт переход на ветку "master", она обновляется, после чего идёт переход на предыдущую ветку и изменения пользователя восстанавливаются.

blrB commented:
а sc-web, рисовалки и scn-editor не обновляем?

Osyotr commented:
@blrB я же в конце вызываю скрипт твой

Osyotr commented:
@blrB понял, где просчитался

Osyotr committed:
scripts: enable updating local branch
The logic is simple: if user has any uncommited changes on his local branch, then the scripts stashes them, checkouts and updates master branch, goes back to users local branch, rebases it with master and finally pops uncommited changes from stash.
